File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> res;
        map<int, int> hashMap;

        for (int eachNum : nums)
        {
            hashMap[eachNum] = eachNum;
        }

        for (int i = 1; i < nums.size() + 1; i++)
        {
            if (hashMap.find(i) == hashMap.end())
            {
                res.push_back(i);
            }
        }
        return res;
    }
};

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:

        hashMap = {}

        for eachNum in nums:
            hashMap[eachNum] = 1

        arr = []
        for eachNum in range(1, len(nums) + 1):
            if eachNum not in hashMap:
                arr.append(eachNum)
        
        return arr